Sources/MapboxNavigation/RecentItem.swift

```diff
@@ -66,12 +70,29 @@ public struct RecentItem: Equatable, Codable {
 
 extension Array where Element == RecentItem {
     
-    public func save() {
-        let encoder = JSONEncoder()
-        let data = try? encoder.encode(self)
-        (try? data?.write(to: RecentItem.filePathUrl)) as ()??
+    /**
+     Method, which allows to save an array of `RecentItem`s into file stored in `recentItemsPathUrl`.
+     */
+    @discardableResult public func save() -> Bool {
+        guard let recentItemsPathUrl = RecentItem.recentItemsPathUrl else { return false }
+        
+        do {
+            let data = try JSONEncoder().encode(self)
+            try data.write(to: recentItemsPathUrl)
+        } catch {
+            NSLog("Failed to save recent items with error: \(error.localizedDescription)")
+            return false
+        }
+        
+        return true
     }
-
+    
+    /**
+     Method, which adds `RecentItem` to the collection. In case if similar `RecentItem` already exists
+     in collection, `timestamp` of its first occurrence will be updated.
+     
+     - parameter recentItem: `RecentItem` instance, which will be added to the collection.
```

Comment:
```suggestion
     Adds a recent item to the collection. If a similar recent item is already in the collection, this method updates the `timestamp` of that item instead of adding a redundant item.
     
     - parameter recentItem: A recent item to add to the collection.
```